File: suc/swift/050.swift
import SwiftUI
import UIKit

// MARK: - 1. AppTokens Definition

extension Color {
    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}

struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0xFF1E3A8A)
        static let secondary = Color(hex: 0xFF2563EB)
        static let tertiary = Color(hex: 0xFF60A5FA)
        static let background = Color(hex: 0xFFF5F8FF)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xFFE8EEFA)
        static let outline = Color(hex: 0xFFCBD5E1)
        static let success = Color(hex: 0xFF16A34A)
        static let warning = Color(hex: 0xFFF59E0B)
        static let error = Color(hex: 0xFFEF4444)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0xFF0B1220)
        static let onBackground = Color(hex: 0xFF0B1220)
        static let onSurface = Color(hex: 0xFF0B1220)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 28, weight: .bold)
        static let headline = Font.system(size: 20, weight: .semibold)
        static let title = Font.system(size: 16, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        static let smallCornerRadius: CGFloat = 8
        static let mediumCornerRadius: CGFloat = 12
        static let largeCornerRadius: CGFloat = 18
    }

    struct Spacing {
        static let xs: CGFloat = 6
        static let sm: CGFloat = 10
        static let md: CGFloat = 14
        static let lg: CGFloat = 18
        static let xl: CGFloat = 26
        static let xxl: CGFloat = 36
    }

    struct ShadowSpec {
        let elevation: CGFloat
        let radius: CGFloat
        let dy: CGFloat
        let opacity: Double
    }

    struct ElevationMapping {
        static let level1 = ShadowSpec(elevation: 2, radius: 4, dy: 2, opacity: 0.12)
        static let level2 = ShadowSpec(elevation: 6, radius: 8, dy: 4, opacity: 0.16)
        static let level3 = ShadowSpec(elevation: 10, radius: 12, dy: 6, opacity: 0.18)
    }
}

// MARK: - 2. Status Bar Hiding Implementation

class HiddenStatusBarHostingController<Content: View>: UIHostingController<Content> {
    override var prefersStatusBarHidden: Bool {
        return true
    }
}

struct HiddenStatusBarView<Content: View>: UIViewControllerRepresentable {
    let content: Content

    func makeUIViewController(context: Context) -> HiddenStatusBarHostingController<Content> {
        return HiddenStatusBarHostingController(rootView: content)
    }

    func updateUIViewController(_ uiViewController: HiddenStatusBarHostingController<Content>, context: Context) {}
}

// MARK: - 3. Custom Segmented Control with Checkmarks

struct SegmentedControl: View {
    @Binding var selectedIndex: Int
    let labels: [String]
    
    var body: some View {
        HStack(spacing: 0) {
            ForEach(0..<labels.count, id: \.self) { index in
                Button(action: {
                    selectedIndex = index
                }) {
                    HStack(spacing: AppTokens.Spacing.xs) {
                        if selectedIndex == index {
                            Image(systemName: "checkmark")
                                .font(.system(size: 14, weight: .semibold))
                                .foregroundColor(AppTokens.Colors.onPrimary)
                        }
                        
                        Text(labels[index])
                            .font(AppTokens.TypographyTokens.label)
                            .foregroundColor(selectedIndex == index ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                    }
                    .frame(maxWidth: .infinity)
                    .frame(height: 40)
                    .background(
                        segmentShape(for: index)
                            .fill(selectedIndex == index ? AppTokens.Colors.primary : AppTokens.Colors.surfaceVariant)
                    )
                }
            }
        }
        .overlay(
            RoundedRectangle(cornerRadius: AppTokens.Shapes.mediumCornerRadius)
                .stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
    }
    
    private func segmentShape(for index: Int) -> RoundedRectangle {
        let radius = AppTokens.Shapes.mediumCornerRadius
        if labels.count == 1 {
            return RoundedRectangle(cornerRadius: radius)
        } else if index == 0 {
            return RoundedRectangle(cornerRadius: radius, style: .continuous)
        } else if index == labels.count - 1 {
            return RoundedRectangle(cornerRadius: radius, style: .continuous)
        } else {
            return RoundedRectangle(cornerRadius: 0)
        }
    }
}

// MARK: - 4. RootScreen with Adjusted Passenger Card Size

struct RootScreen: View {
    @State private var name: String = ""
    @State private var code: String = ""
    @State private var selectedSeat: Int = 0
    @State private var showDialog: Bool = false

    let seatTabs = ["Economy", "Premium", "Business"]

    var body: some View {
        ZStack {
            LinearGradient(
                gradient: Gradient(colors: [
                    AppTokens.Colors.tertiary.opacity(0.25),
                    AppTokens.Colors.background,
                    AppTokens.Colors.secondary.opacity(0.2)
                ]),
                startPoint: .top,
                endPoint: .bottom
            )
            .ignoresSafeArea(.all)

            VStack(spacing: 0) {
                HStack {
                    Circle()
                        .fill(AppTokens.Colors.primary)
                        .frame(width: 22, height: 22)
                        .padding(.leading, AppTokens.Spacing.lg)

                    Spacer()

                    Text("Flight Check-in")
                        .font(AppTokens.TypographyTokens.display)
                        .foregroundColor(AppTokens.Colors.onSurface)

                    Spacer()

                    Circle()
                        .fill(AppTokens.Colors.secondary)
                        .frame(width: 22, height: 22)
                        .padding(.trailing, AppTokens.Spacing.lg)
                }
                .frame(height: 56)
                .padding(.bottom, AppTokens.Spacing.md)

                ScrollView {
                    VStack(spacing: AppTokens.Spacing.lg) {
                        // Passenger & Booking Card - 增大这个卡片
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.lg) { // 增加内部间距
                            Text("Passenger")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)

                            TextField("Full name", text: $name)
                                .font(AppTokens.TypographyTokens.body)
                                .padding(AppTokens.Spacing.md)
                                .background(AppTokens.Colors.surfaceVariant)
                                .cornerRadius(AppTokens.Shapes.mediumCornerRadius)
                                .foregroundColor(AppTokens.Colors.onSurface)
                                .overlay(
                                    RoundedRectangle(cornerRadius: AppTokens.Shapes.mediumCornerRadius)
                                        .stroke(AppTokens.Colors.outline, lineWidth: 1)
                                )

                            Text("Booking Code")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)

                            TextField("e.g. XY7ZK2", text: $code)
                                .font(AppTokens.TypographyTokens.body)
                                .padding(AppTokens.Spacing.md)
                                .background(AppTokens.Colors.surfaceVariant)
                                .cornerRadius(AppTokens.Shapes.mediumCornerRadius)
                                .foregroundColor(AppTokens.Colors.onSurface)
                                .overlay(
                                    RoundedRectangle(cornerRadius: AppTokens.Shapes.mediumCornerRadius)
                                        .stroke(AppTokens.Colors.outline, lineWidth: 1)
                                )

                            Text("Seat Class")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)

                            SegmentedControl(selectedIndex: $selectedSeat, labels: seatTabs)
                        }
                        .padding(AppTokens.Spacing.xl) // 增加内边距使卡片更大
                        .background(AppTokens.Colors.surface)
                        .cornerRadius(AppTokens.Shapes.largeCornerRadius)
                        .shadow(color: .black.opacity(AppTokens.ElevationMapping.level2.opacity),
                                radius: AppTokens.ElevationMapping.level2.radius,
                                x: 0,
                                y: AppTokens.ElevationMapping.level2.dy)

                        // Flight Details Card
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.md) {
                            // Flight信息行 - 添加白色背景
                            HStack {
                                VStack(alignment: .leading, spacing: AppTokens.Spacing.xs) {
                                    Text("Flight")
                                        .font(AppTokens.TypographyTokens.title)
                                        .foregroundColor(AppTokens.Colors.onSurface)
                                    Text("BL 213 • HND → CTS")
                                        .font(AppTokens.TypographyTokens.body)
                                        .foregroundColor(AppTokens.Colors.onSurface)
                                }
                                Spacer()
                                Circle()
                                    .fill(AppTokens.Colors.secondary)
                                    .frame(width: 36, height: 36)
                            }
                            .padding(AppTokens.Spacing.md)
                            .background(AppTokens.Colors.surface)
                            .cornerRadius(AppTokens.Shapes.mediumCornerRadius)

                            HStack {
                                VStack(alignment: .leading) {
                                    Text("Boarding")
                                        .font(AppTokens.TypographyTokens.label)
                                        .foregroundColor(AppTokens.Colors.onSurface)
                                    Text("12:35")
                                        .font(AppTokens.TypographyTokens.title)
                                        .foregroundColor(AppTokens.Colors.primary)
                                }
                                Spacer()
                                VStack(alignment: .leading) {
                                    Text("Gate")
                                        .font(AppTokens.TypographyTokens.label)
                                        .foregroundColor(AppTokens.Colors.onSurface)
                                    Text("A12")
                                        .font(AppTokens.TypographyTokens.title)
                                        .foregroundColor(AppTokens.Colors.primary)
                                }
                                Spacer()
                                VStack(alignment: .leading) {
                                    Text("Seat")
                                        .font(AppTokens.TypographyTokens.label)
                                        .foregroundColor(AppTokens.Colors.onSurface)
                                    Text(seatNumber(for: selectedSeat))
                                        .font(AppTokens.TypographyTokens.title)
                                        .foregroundColor(AppTokens.Colors.primary)
                                }
                            }
                            .padding(.vertical, AppTokens.Spacing.md)

                            Button(action: {
                                showDialog = true
                            }) {
                                Text("Confirm Check-in")
                                    .font(AppTokens.TypographyTokens.title)
                                    .frame(maxWidth: .infinity)
                                    .frame(height: 52)
                                    .background(AppTokens.Colors.primary)
                                    .foregroundColor(AppTokens.Colors.onPrimary)
                                    .cornerRadius(AppTokens.Shapes.largeCornerRadius)
                            }
                        }
                        .padding(AppTokens.Spacing.lg)
                        .background(AppTokens.Colors.surface)
                        .cornerRadius(AppTokens.Shapes.largeCornerRadius)
                        .shadow(color: .black.opacity(AppTokens.ElevationMapping.level2.opacity),
                                radius: AppTokens.ElevationMapping.level2.radius,
                                x: 0,
                                y: AppTokens.ElevationMapping.level2.dy)
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .padding(.vertical, AppTokens.Spacing.md)
                }
            }
        }
        .alert("Check-in Successful", isPresented: $showDialog) {
            Button("Done") {
                showDialog = false
            }
        } message: {
            Text("Boarding pass is ready. Have a nice flight.")
        }
    }

    private func seatNumber(for index: Int) -> String {
        if index == 0 {
            return "23C"
        } else if index == 1 {
            return "11A"
        } else {
            return "2F"
        }
    }
}

// MARK: - 5. App Entry Point

@main
struct FlightCheckinApp: App {
    var body: some Scene {
        WindowGroup {
            HiddenStatusBarView(content: RootScreen())
        }
    }
}

File: suc/kotlin/050.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.SegmentedButton
import androidx.compose.material3.SegmentedButtonDefaults
import androidx.compose.material3.SingleChoiceSegmentedButtonRow
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "065FlightCheckinen"
private const val UI_TYPE = "Travel"
private const val STYLE_THEME = "Modern Blue"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF1E3A8A)
        val secondary = Color(0xFF2563EB)
        val tertiary = Color(0xFF60A5FA)
        val background = Color(0xFFF5F8FF)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE8EEFA)
        val outline = Color(0xFFCBD5E1)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFF0B1220)
        val onBackground = Color(0xFF0B1220)
        val onSurface = Color(0xFF0B1220)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val headline = TextStyle(fontSize = 20.sp, fontWeight = FontWeight.SemiBold)
        val title = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(8.dp)
        val medium = RoundedCornerShape(12.dp)
        val large = RoundedCornerShape(18.dp)
    }
    object Spacing {
        val xs = 6.dp
        val sm = 10.dp
        val md = 14.dp
        val lg = 18.dp
        val xl = 26.dp
        val xxl = 36.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f)
        val level2 = ShadowSpec(6.dp, 8.dp, 4.dp, 0.16f)
        val level3 = ShadowSpec(10.dp, 12.dp, 6.dp, 0.18f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val name = remember { mutableStateOf("") }
    val code = remember { mutableStateOf("") }
    val seatTabs = listOf("Economy", "Premium", "Business")
    val selectedSeat = remember { mutableIntStateOf(0) }
    val showDialog = remember { mutableStateOf(false) }
    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Flight Check-in", style = MaterialTheme.typography.displayLarge, color = MaterialTheme.colorScheme.onSurface) },
                navigationIcon = {
                    Box(modifier = Modifier.padding(start = AppTokens.Spacing.lg).size(22.dp).background(MaterialTheme.colorScheme.primary, CircleShape))
                },
                actions = {
                    Box(modifier = Modifier.padding(end = AppTokens.Spacing.lg).size(22.dp).background(MaterialTheme.colorScheme.secondary, CircleShape))
                }
            )
        },
        containerColor = Color.Transparent
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .background(
                    Brush.verticalGradient(
                        listOf(
                            AppTokens.Colors.tertiary.copy(alpha = 0.25f),
                            AppTokens.Colors.background,
                            AppTokens.Colors.secondary.copy(alpha = 0.2f)
                        )
                    )
                )
                .padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            Card(
                shape = AppTokens.Shapes.large,
                colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                elevation = CardDefaults.cardElevation(AppTokens.ElevationMapping.level2.elevation),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg),
                    verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
                ) {
                    Text("Passenger", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    TextField(
                        value = name.value,
                        onValueChange = { name.value = it },
                        modifier = Modifier.fillMaxWidth(),
                        placeholder = { Text("Full name", color = AppTokens.Colors.onSurface.copy(alpha = 0.5f)) },
                        colors = TextFieldDefaults.colors(
                            focusedContainerColor = AppTokens.Colors.surfaceVariant,
                            unfocusedContainerColor = AppTokens.Colors.surfaceVariant,
                            focusedIndicatorColor = AppTokens.Colors.primary,
                            unfocusedIndicatorColor = AppTokens.Colors.outline
                        ),
                        shape = AppTokens.Shapes.medium
                    )
                    Text("Booking Code", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    TextField(
                        value = code.value,
                        onValueChange = { code.value = it },
                        modifier = Modifier.fillMaxWidth(),
                        placeholder = { Text("e.g. XY7ZK2", color = AppTokens.Colors.onSurface.copy(alpha = 0.5f)) },
                        colors = TextFieldDefaults.colors(
                            focusedContainerColor = AppTokens.Colors.surfaceVariant,
                            unfocusedContainerColor = AppTokens.Colors.surfaceVariant,
                            focusedIndicatorColor = AppTokens.Colors.primary,
                            unfocusedIndicatorColor = AppTokens.Colors.outline
                        ),
                        shape = AppTokens.Shapes.medium
                    )
                    Text("Seat Class", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    SingleChoiceSegmentedButtonRow {
                        seatTabs.forEachIndexed { index, label ->
                            SegmentedButton(
                                selected = selectedSeat.intValue == index,
                                onClick = { selectedSeat.intValue = index },
                                shape = SegmentedButtonDefaults.itemShape(index, seatTabs.size),
                                colors = SegmentedButtonDefaults.colors(
                                    activeContainerColor = AppTokens.Colors.primary,
                                    activeContentColor = AppTokens.Colors.onPrimary,
                                    inactiveContainerColor = AppTokens.Colors.surfaceVariant,
                                    inactiveContentColor = AppTokens.Colors.onSurface
                                )
                            ) {
                                Text(label, style = AppTokens.TypographyTokens.label)
                            }
                        }
                    }
                }
            }
            Card(
                shape = AppTokens.Shapes.large,
                colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                elevation = CardDefaults.cardElevation(AppTokens.ElevationMapping.level2.elevation),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg),
                    verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column(verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)) {
                            Text("Flight", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.onSurface)
                            Text("BL 213 • HND → CTS", style = AppTokens.TypographyTokens.body, color = AppTokens.Colors.onSurface)
                        }
                        Box(modifier = Modifier.size(36.dp).background(AppTokens.Colors.secondary, CircleShape))
                    }
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text("Boarding", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.onSurface)
                            Text("12:35", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.primary)
                        }
                        Column {
                            Text("Gate", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.onSurface)
                            Text("A12", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.primary)
                        }
                        Column {
                            Text("Seat", style = AppTokens.TypographyTokens.label, color = AppTokens.Colors.onSurface)
                            Text(if (selectedSeat.intValue == 0) "23C" else if (selectedSeat.intValue == 1) "11A" else "2F", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.primary)
                        }
                    }
                    Button(
                        onClick = { showDialog.value = true },
                        colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary, contentColor = AppTokens.Colors.onPrimary),
                        shape = AppTokens.Shapes.large,
                        modifier = Modifier.fillMaxWidth().height(52.dp)
                    ) {
                        Text("Confirm Check-in", style = AppTokens.TypographyTokens.title)
                    }
                }
            }
        }
    }
    if (showDialog.value) {
        AlertDialog(
            onDismissRequest = { showDialog.value = false },
            title = { Text("Check-in Successful", style = AppTokens.TypographyTokens.title, color = AppTokens.Colors.onSurface) },
            text = { Text("Boarding pass is ready. Have a nice flight.", style = AppTokens.TypographyTokens.body, color = AppTokens.Colors.onSurface) },
            confirmButton = {
                Button(
                    onClick = { showDialog.value = false },
                    colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.secondary, contentColor = AppTokens.Colors.onSecondary),
                    shape = AppTokens.Shapes.medium
                ) { Text("Done", style = AppTokens.TypographyTokens.label) }
            }
        )
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF5F8FF)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}